src/ORM/ListDecorator.php

```diff
@@ -16,21 +16,16 @@ abstract class ListDecorator extends ViewableData implements SS_List, Sortable,
     /**
      * @var SS_List
      */
-    protected $list;
+    protected SS_List&Sortable&Filterable&Limitable $list;
```

Comment:
Basically, everything method on the class assumes that we will return a List that implements all these interfaces.